### PR TITLE
fix: file transfer

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Several fix for the file transfer project:
- Fix the application yaml usage
- Remove handle batch for FileObject -> Everything is done in the process records
- Fix serde of destination file.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/write/object_storage/ObjectStorageStreamLoaderFactory.kt [hunk 1]
```diff
@@ -117,16 +113,27 @@ class ObjectStorageStreamLoader<T : RemoteObject<*>, U : OutputStream>(
     }
 
     override suspend fun processFile(file: DestinationFile): Batch {
+        if (pathFactory.supportsStaging) {
+            throw IllegalStateException("Staging is not supported for files")
```
Comment (johnny-schmidt) [inline airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/write/object_storage/ObjectStorageStreamLoaderFactory.kt:117]: Can we add this to the `--check`? (See `S3V2Checker`.)

**Review by johnny-schmidt — APPROVED** [review]
Minor nits, but lgtm

File: airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/task/MockTaskLauncher.kt [hunk 2]
```diff
@@ -49,7 +49,11 @@ class MockTaskLauncher : DestinationTaskLauncher {
         throw NotImplementedError()
     }
 
-    override suspend fun handleFile(stream: DestinationStream.Descriptor, file: DestinationFile) {
+    override suspend fun handleFile(
+        stream: DestinationStream.Descriptor,
+        file: DestinationFile,
+        index: Long
+    ) {
         throw NotImplementedError()
```
[the remark this answers is not in the document]
Comment (benmoriceau) [reply airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/task/MockTaskLauncher.kt]: Done

File: airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/internal/InputConsumerTask.kt [hunk 3]
```diff
@@ -88,10 +91,14 @@ class DefaultInputConsumerTask(
...
                 throw IllegalStateException("Stream $stream failed upstream, cannot continue.")
             is DestinationFile -> {
-                destinationTaskLauncher.handleFile(stream, message)
+                val index = manager.countRecordIn()
+                destinationTaskLauncher.handleFile(stream, message, index)
             }
             is DestinationFileStreamComplete -> {
                 reserved.release() // safe because multiple calls conflate
+                manager.markEndOfStream()
+                val envelope = BatchEnvelope(SimpleBatch(Batch.State.COMPLETE))
```
Comment (tryangul) [inline airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/internal/InputConsumerTask.kt:100]: Can we leave a comment about why we have to create a dummy batch here?

**Review by tryangul — APPROVED** [review]

**Pull request merged**